Makefile

```diff
@@ -0,0 +1,221 @@
+SHELL := /bin/bash
+.DEFAULT_GOAL := help
+MAKE_NPROCS ?= $(shell nproc)
+CREME_LANGUAGE ?= fr
+
+
+## Upgrade the Python requirements
+.PHONY: update-requirements
+update-requirements:
+	pip install --upgrade -r creme/requirements-dev.txt
+
+
+## Upgrade the Python requirements, run the migrations, the creme_populate and generatemedia commands
+.PHONY: update
+update: update-requirements
+	python manage.py migrate
+	python manage.py creme_populate
+	python manage.py generatemedia
+
+
+## Generate the media files
+.PHONY: media
+media:
+	python manage.py generatemedia
+
+
+## Run the Django test suite
+.PHONY: test
+test:
+	python manage.py test --noinput --parallel=${MAKE_NPROCS} $(filter-out $@,$(MAKECMDGOALS))
+
+
+## Run the Django test suite and generate coverage reports
+.PHONY: test-cov
+test-cov:
+	COVERAGE_PROCESS_START=.coveragerc coverage run --source creme/ manage.py test --noinput --parallel=${MAKE_NPROCS} $(filter-out $@,$(MAKECMDGOALS))
+	coverage combine
+	coverage report
+	coverage html
+
+
+## Run the Javascript test suite
+.PHONY: karma
+karma: media
+	node_modules/.bin/karma start .karma.conf.js --browsers=FirefoxHeadless --targets=$(filter-out $@,$(MAKECMDGOALS))
+
+
+## Run the application
+.PHONY: serve
+serve: media
+	python manage.py runserver
+
+
+## Run the Javascript linters
+.PHONY: eslint
+eslint:
+	git diff --name-only origin/master creme/ | { grep '.js$$' || true; } | xargs --no-run-if-empty \
+		node_modules/.bin/eslint \
+			--config .eslintrc \
+			--ignore-path .eslintignore \
+			--format stylish \
+			--quiet
+
+
+## Validates the Python imports with isort
+.PHONY: isort-check
+isort-check:
+	git diff --name-only origin/master creme/ | { grep '.py$$' || true; } | xargs --no-run-if-empty \
+		isort --check --diff --atomic --verbose
+
+
+## Sort the Python imports with isort
+.PHONY: isort-fix
+isort-fix:
+	git diff --name-only origin/master creme/ | { grep '.py$$' || true; } | xargs --no-run-if-empty \
+		isort --atomic --verbose
+
+
+## Validates the Python code with flake8
+.PHONY: flake8
+flake8:
+	flake8 creme
+
+
+## Run all the Python linters
+.PHONY: lint
+lint: isort-check flake8
+
+
+## Collect the messages to translate
+.PHONY: gettext-collect
+gettext-collect:
+	pushd ./creme/products && \
```

Comment:
I learned (or learn again) something today, thx :smile:
But I would preferred a loop and ignore arguments for all usecases like `-i */tests/* -i */tests.py -i creme/*.py`
Maybe a little `gettextcollect' command in python is a better idea for this : allows simplier language/app selection, handling common & local "ignore" files, etc...
Or using pybabel :smile:

Reply:
Well, I agree this can still be improved, but it has the merit of doing its job and of beeing non-invasive. I suggest we iterate on it in a next PR, as it's not about reworking the translation system, but provide tools to help us work faster.

Reply:
Ok, but it is not really usable with client apps